File: algorithms/c#/SelectionSort.cs
namespace Algorithms
{
	public static class SelectionSort
	{
		public static void Sort(int[] array)
		{
			int min = 0;

			for(int i = 0; i < array.Length; i++)
			{
				min = i;

				for(int j = i + 1; j < array.Length; j++)
				{
					if(array[j] < array[min]) min = j;
				}

				int a = array[i];
				int b = array[min];
				array[i] = b;
				array[min] = a;
			}
		}
	}
}

File: algorithms/go/selection_sort.go
package algorithms

func SelectionSort(slice []int) {
	min := 0

	for i := 0; i < len(slice); i++ {
		min = i

		for j := i + 1; j < len(slice); j++ {
			if slice[j] < slice[min] {
				min = j;
			}
		}

		temp1 := slice[i]
		temp2 := slice[min]

		slice[i] = temp2
		slice[min] = temp1
	}
}